source/extensions/filters/http/jwt_authn/authenticator.cc

```diff
@@ -169,6 +169,12 @@ void AuthenticatorImpl::startVerify() {
   // isIssuerSpecified() check already make sure the issuer is in the cache.
   ASSERT(jwks_data_ != nullptr);
 
+  if (jwks_data_->getTokenCache()->lookupTokenCache(curr_token_->token())) {
+    ENVOY_LOG(debug, "{}: verified token with cache hit: tokens size {}", name(), tokens_.size());
+    doneWithStatus(Status::Ok);
```

Comment:
Do we need to use cached jwt_?

Reply:
No, I don't think it's required.

Reply:
it is needed. It needs to be used in line 268 and line 259.  
If a jwt is good, its payload need to write to headers and metadata.  We should do that even for a cached jwt.

Reply:
Done.